basic-zookeeper-aware-server/src/main/java/com/rtbhouse/grpc/loadbalancer/zookeeper/ZooKeeperAwareBasicGrpcServer.java

```diff
@@ -0,0 +1,142 @@
+package com.rtbhouse.grpc.loadbalancer.zookeeper;
+
+import static com.rtbhouse.grpc.loadbalancer.Helper.getServerAddress;
+
+import com.rtbhouse.grpc.loadbalancer.HealthCheckCapableGrpcServerBase;
+import io.grpc.BindableService;
+import io.grpc.health.v1.HealthGrpc.HealthImplBase;
+import java.io.IOException;
+import java.net.InetAddress;
+import org.apache.curator.framework.CuratorFramework;
+import org.apache.curator.framework.imps.CuratorFrameworkState;
+import org.apache.curator.framework.state.ConnectionState;
+import org.apache.curator.framework.state.ConnectionStateListener;
+import org.apache.zookeeper.CreateMode;
+import org.apache.zookeeper.KeeperException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class ZooKeeperAwareBasicGrpcServer extends HealthCheckCapableGrpcServerBase {
+  private static final Logger logger = LoggerFactory.getLogger(ZooKeeperAwareBasicGrpcServer.class);
+  private final String zooNodePath;
+  private final CuratorFramework curatorFramework;
+  private final boolean usingNAT;
+
+  private volatile ConnectionState connectionState;
+
+  private ConnectionStateListener serverNodeRecreateListener =
+      (client, newState) -> {
+        connectionState = newState;
+        if (connectionState.isConnected()) {
+          if (!super.checkingHealth || isHealthy()) {
+            registerServerNode();
+          }
+        }
+      };
+
+  public ZooKeeperAwareBasicGrpcServer(
```

Comment:
Ta klasa też mogłaby mieć buildera

Reply:
Prawda - ale przez to że skupiamy się teraz mocno na 3ciej, rozproszonej wersji to nie chcemy jeszcze poświęcać czasu na poprawki w zookeeperze, on tak naprawdę był taką rozgrzewką i POC dla nas że to faktycznie działa. Chyba w ogóle na razie usunę tę część z pull requesta, bo dużo, dużo ważniejsza jest 2ga iteracja (tzn. loadbalancer-standalone) a tak to ta część zookeeperowa niepotrzebnie zaciemnia i utrudnia reviewowanie.